presto-main/src/main/java/com/facebook/presto/operator/exchange/LocalExchange.java

```diff
@@ -355,16 +436,13 @@ else if (partitioning.equals(FIXED_ARBITRARY_DISTRIBUTION)) {
             bufferCount = defaultConcurrency;
             checkArgument(partitionChannels.isEmpty(), "Arbitrary exchange must not have partition channels");
         }
-        else if (partitioning.equals(FIXED_HASH_DISTRIBUTION)) {
-            bufferCount = defaultConcurrency;
-            checkArgument(!partitionChannels.isEmpty(), "Partitioned exchange must have partition channels");
-        }
         else if (partitioning.equals(FIXED_PASSTHROUGH_DISTRIBUTION)) {
             bufferCount = defaultConcurrency;
             checkArgument(partitionChannels.isEmpty(), "Passthrough exchange must not have partition channels");
         }
         else {
-            throw new IllegalArgumentException("Unsupported local exchange partitioning " + partitioning);
+            // partitioned exchange
```

Comment:
ditto: add a check it either has to be connector partitioning, or FIXED_HASH_DISTRIBUTION